### PR TITLE
Resolved two Issue #389 #396

### PR DESCRIPTION
**Pull request by gdmmx**
### Proposed changes in this pull request
Resolved Issue #389 
Resolved Issue #396

### Type (put an `x` where ever applicable)
- [x] Bug fix: Link to the issue
- [x] Feature (Non-breaking change)
- [ ] Feature (Breaking change)
- [ ] Documentation Improvement

### Checklist
Please put an `x` against the checkboxes. Write a small comment explaining if its `N/A` (not applicable)

- [x] Read the [CONTRIBUTION guidelines](https://github.com/nknorg/nkn#contributing).
- [x] All the tests are passing after the introduction of new changes.
- [ ] Added tests respective to the part of code I have written.
- [ ] Added proper documentation where ever applicable (in code and README.md).
- [x] Code has been written according to [NKN-Golang-Style-Guide](https://github.com/nknorg/nkn/wiki/NKN-Golang-Style-Guide)

### Extra information
Any extra information related to this pull request.

### CODE REVIEW

**Comment by yilunzhang** [comment]
@CrackDavid

**Comment by gdmmx** [comment]
BTW, @CrackDavid You can get correct Block Height via getnodestate API now.
The data type of version field still not changed yet since it is not compatible with old version online node. (About Issue #372 )

File: crypto/encode.go [hunk 1]
```diff
@@ -312,8 +312,5 @@ func NewPubKey(priKey []byte) *PubKey {
 
 	privateKey.PublicKey.X, privateKey.PublicKey.Y = algSet.Curve.ScalarBaseMult(k.Bytes())
 
-	pubKey := new(PubKey)
-	pubKey.X = privateKey.PublicKey.X
-	pubKey.Y = privateKey.PublicKey.Y
-	return pubKey
+	return &PubKey{X: privateKey.PublicKey.X, Y: privateKey.PublicKey.Y}
```
Comment (gdmmx) [inline crypto/encode.go]: Code changed just for code refined

**Comment by CrackDavid** [comment]
> BTW, @CrackDavid You can get correct Block Height via getnodestate API now.
> The data type of version field still not changed yet since it is not compatible with old version online node. (About Issue #372 )

Very good, thanks for the information!

**Pull request merged**